modules/network/rules.tf

```diff
@@ -113,6 +113,29 @@ resource "oci_core_network_security_group_security_rule" "oke" {
       condition     = each.value.direction == "INGRESS" || coalesce(each.value.destination, "none") != "none"
       error_message = "Egress rule must have a destination: '${each.key}'"
     }
+
```

Comment:
On line 91:

`
    precondition {
      condition = tostring(each.value.protocol) == tostring(local.icmp_protocol) || contains(keys(each.value), "port") || (
        contains(keys(each.value), "port_min") && contains(keys(each.value), "port_max")
      )
      error_message = "TCP/UDP rule must contain a port or port range: '${each.key}'"
    }
`
Shouldn't this be checking for `local.tcp_protocol` or `local.udp_protocol`?

Reply:
This is correct: the rule gets a pass if it's icmp; otherwise it must have either `port` or (`port_min` and `port_max`).

Reply:
Ok, please document the code.